commands.txt

```diff
@@ -1,4 +1,4 @@
 if [ `uname` = 'Darwin' ]; then export SD_CURL_CMD_WRAPPER="eval" && export SD_ZIP_CMD_WRAPPER=""; else export SD_CURL_CMD_WRAPPER="$(echo 'sd-step exec --pkg-version 7.54.1 core/curl')" && export SD_ZIP_CMD_WRAPPER="$(echo 'sd-step exec core/zip')"; fi
 cd $SD_ARTIFACTS_DIR
 find . -type f -print > manifest.txt
-if [ "$SD_ZIP_ARTIFACTS" = "true" ]; then ZIPFILE=SD_ARTIFACT.zip && find . -type f -exec $SD_ZIP_CMD_WRAPPER zip $ZIPFILE {} + && f=`echo $ZIPFILE | sed "s/^\.\///"`; encoded=$($SD_CURL_CMD_WRAPPER "curl -s -w \"%{url_effective}\" --data-urlencode \"$f\" -G \"\" | sed -e \"s/^\/?//\"") && $SD_CURL_CMD_WRAPPER "curl -H \"Authorization: Bearer $SD_TOKEN\" -H \"Content-Type: text/plain\" -X PUT $STORE_URL/v1/builds/$SD_BUILD_ID/$ARTIFACTS_DIR_SUFFIX/$encoded -T ./\"$f\""; else find . -type f -exec sh -c 'f=`echo $1 | sed "s/^\.\///"`; encoded=$($SD_CURL_CMD_WRAPPER "curl -s -w \"%{url_effective}\" --data-urlencode \"$f\" -G \"\" | sed -e \"s/^\/?//\""); $SD_CURL_CMD_WRAPPER "curl -H \"Authorization: Bearer $SD_TOKEN\" -H \"Content-Type: text/plain\" -X PUT $STORE_URL/v1/builds/$SD_BUILD_ID/$ARTIFACTS_DIR_SUFFIX/$encoded -T ./\"$f\""' -- {} \;; fi
+if [ "$SD_ZIP_ARTIFACTS" = "true" ]; then ZIPFILE=SD_ARTIFACT.zip && find . -type f -exec $SD_ZIP_CMD_WRAPPER zip $ZIPFILE {} + && f=`echo $ZIPFILE | sed "s/^\.\///"`; encoded=$($SD_CURL_CMD_WRAPPER "curl -s -w \"%{url_effective}\" --data-urlencode \"$f\" -G \"\" | sed -e \"s/^\/?//\"") && $SD_CURL_CMD_WRAPPER "curl --retry 3 -H \"Authorization: Bearer $SD_TOKEN\" -H \"Content-Type: text/plain\" -X PUT $STORE_URL/v1/builds/$SD_BUILD_ID/$ARTIFACTS_DIR_SUFFIX/$encoded -T ./\"$f\""; else find . -type f -exec sh -c 'f=`echo $1 | sed "s/^\.\///"`; encoded=$($SD_CURL_CMD_WRAPPER "curl -s -w \"%{url_effective}\" --data-urlencode \"$f\" -G \"\" | sed -e \"s/^\/?//\""); $SD_CURL_CMD_WRAPPER "curl --retry 3 -H \"Authorization: Bearer $SD_TOKEN\" -H \"Content-Type: text/plain\" -X PUT $STORE_URL/v1/builds/$SD_BUILD_ID/$ARTIFACTS_DIR_SUFFIX/$encoded -T ./\"$f\""' -- {} \;; fi
```

Comment:
can we also have a delay between retries ? Does `curl` exposes a config for that ?

Reply:
It has a default `retry-delay` of 1 second for the first retry, 2 for the second, 4 for the third, and it exponentially increases with each retry.